core/dbt/artifacts/resources/v1/macro.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass, field
+import time
+from typing import Literal, List, Dict, Optional, Any
+
+from dbt_common.contracts.util import Replaceable
+from dbt_common.dataclass_schema import dbtClassMixin
+from dbt.artifacts.resources.base import BaseArtifactNode
+from dbt.artifacts.resources.types import NodeType, ModelLanguage
+from dbt.artifacts.resources.v1.docs import Docs
+
+
+@dataclass
+class MacroArgument(dbtClassMixin):
+    name: str
+    type: Optional[str] = None
+    description: str = ""
+
+
+@dataclass
+class MacroDependsOn(dbtClassMixin, Replaceable):
+    macros: List[str] = field(default_factory=list)
+
+    # 'in' on lists is O(n) so this is O(n^2) for # of macros
+    def add_macro(self, value: str):
```

Comment:
Went back and forth on whether this method belongs in dbt/artifacts given it is mutating / not read-only. An alternative would be to define the method more directly on the internal `Macro` node class. 

 Decided to leave it as-is for now because `DependsOn` inherits from `MacroDependsOn`. We should revisit once we move `DependsOn` though.

Reply:
For those looking in @MichelleArk and I have talked about this a bit in DMs. In my PR #9460 I plan on moving `DependsOn` in full over to artifacts. We plan in the future to revist the design of both `MacroDependsOn` and `DependsOn` to perhaps simplify and improve them.